src/doc_builder/autodoc.py

```diff
@@ -414,14 +416,25 @@ def autodoc(object_name, package, methods=None, return_anchors=False, page_info=
             methods_to_add = find_documented_methods(obj)
             methods.extend([m for m in methods_to_add if m not in methods])
         for method in methods:
+            anchor_name = f"{anchors[0]}.{method}"
             method_doc, check = document_object(
-                object_name=f"{object_name}.{method}", package=package, page_info=page_info, full_name=False
+                object_name=f"{object_name}.{method}",
+                package=package,
+                page_info=page_info,
+                full_name=False,
+                anchor_name=anchor_name,
             )
             if check is not None:
                 errors.append(check)
             documentation += '\n<div class="docstring">' + method_doc + "</div>"
             if return_anchors:
-                anchors.append(f"{anchors[0]}.{method}")
+                # The anchor name of the method might be different from its
+                method = find_object_in_package(f"{anchors[0]}.{method}", package=package)
+                method_name = get_shortest_path(method, package=package)
```

Comment:
This replicates what the doc-builder will do when we try to find we try to resolve link to ``[`BertTokenizer.__call__`]`` -> it will get a different `method_name` compared to the `anchor_name`.